warehouse/bigquery.go

```diff
@@ -206,26 +205,29 @@ func (bq *BigQuery) EnsureCompatibleExportTable() error {
 		return err
 	}
 
+	needsUpdate := false
 	// Find the fields from the hauser schema that are missing from the BiqQuery table
 	missingFields := bq.GetMissingFields(hauserSchema, md.Schema)
-
 	// If fields are missing, we add them to the table schema
+	update := bigquery.TableMetadataToUpdate{}
 	if len(missingFields) > 0 {
 		// Append missing fields to export table schema
-		md.Schema = append(md.Schema, missingFields...)
-		if err := bq.updateTable(table, md.Schema); err != nil {
-			return nil
+		update.Schema = append(md.Schema, missingFields...)
+		needsUpdate = true
+	}
+
+	if md.TimePartitioning.Expiration != bq.conf.BigQuery.PartitionExpiration.Duration {
+		update.TimePartitioning = &bigquery.TimePartitioning{
+			Expiration: bq.conf.BigQuery.PartitionExpiration.Duration,
+			Field:      md.TimePartitioning.Field,
```

Comment:
Now should the field be set based on the export data? Like by using `SessionStart`?

Reply:
When hauser was created, you couldn't partition a table in BQ using a field value, it was always partitioned by ingestion time. I think that's how the tables are still being created by hauser today. So in `LoadToWarehouse` we are doing some super hacky stuff with the legacy notation to load data into a particular partition. We had a card in at some point to change to creating the table such that it partitions by the `EventStart` field, which is semantically what our existing hack does now.

I think carrying over the partitioning field from the loaded table metadata here sets us up for a smooth transition to however we might change partitioning in the future.

Reply:
> I think carrying over the partitioning field from the loaded table metadata here sets us up for a smooth transition to however we might change partitioning in the future.

Yeah @jameremo that's what I was thinking. I was intending on following up with that change.

Reply:
Gotcha. Yeah, Cam shared a card with me directly. SGTM